### DISCUSSION TITLE
Adjust Humidity Badge Color - Red = Bad, Green = good

### DISCUSSION THREAD

**Discussion by rsocko** [WolfwithSword/Bambu-HomeAssistant-Flows#19] [Ideas]
I made an adjustment to the AMS yaml configuration to alter the badge outline color for the AMS Humidity sensor.

The default seemed to use reverse logic to what I would have expected. The original shows the badge outline as yellow for any value > 3 (so when higher humidity) and red colors from 1-3 with brightest red at value of 1.

This way if I have a low humidity (eg 1) then it is green as a good value. Only if the humidity is higher 3 or more, then it shows red or error-color.

I adjusted it so that it adds a color level and shows the following: 

Humidity Value | Color: 
1 = Green
2 = Yellow
3 or 4 = Light Red
5 = Red

My change is below: 

```
      --humidity-border-color: {% if states('sensor.ntk_ryansoffice_3dprinterams1_humidity_index')  != 'unavailable' and states('sensor.ntk_ryansoffice_3dprinterams1_humidity_index') | int > 3 %} rgba(194,74,72,1.0); {% elif states('sensor.ntk_ryansoffice_3dprinterams1_humidity_index')  != 'unavailable' and states('sensor.ntk_ryansoffice_3dprinterams1_humidity_index') | int > 2 %} rgba(228,127,97,1.0); {% elif states('sensor.ntk_ryansoffice_3dprinterams1_humidity_index')  != 'unavailable' and states('sensor.ntk_ryansoffice_3dprinterams1_humidity_index') | int > 1 %} rgba(255, 255, 126, 0.5); {% else %} rgba(0,255,17,0.75); {% endif %}

```

I would be happy to submit a PR if it makes sense, but wanted to share.

**Comment by WolfwithSword** [comment]
@rsocko  could you mention which version of the AMS YAML you are using? NodeRed or HACs? 

The problem is, the data from MQTT is a bit weird - as in, it reports 1-5, but it's actually inverted to what the icons are from say, bambustudio. A level 5 is actually a 1, and a level 1 is actually a 5. 

In my NodeRed data I have corrected for this so the icons aren't as confusing. The icon with a "1" is "good", as it is lowest humidity, and the icon with a "5" is worst, as it is high humidity and thus should be red.

I thought I had seen the HACs data also corrected for this, but if not, could you please provide some examples of the actual entity/sensor value from HACs (if you are using HACs) and then a second screenshot of the currently displayed icon and colour? If it is the case the HACs integration has not inverted them, then there will be more changes than just the colours, but also the logic behind fetching the correct icon entirely (as it relies on using the integer value in the name, it will now require some math inside to invert it).

**Comment by WolfwithSword** [comment]
Reconfirmed with HACs devs, they also invert the value. So value of 1 being green is good. But also seeing now at some point the logic was indeed messed up in the yaml. I'll fix it up for the next update.
